xxxx-ledger-plugin-interface-2/xxxx-ledger-plugin-interface-2.md

```diff
@@ -0,0 +1,502 @@
+
+This method MAY reject with any arbitrary JavaScript error.
+
+###### Example
+```js
+p.sendTransfer({
+  amount: '10',
+  destination: 'example.us.acmebank.bob',
+  executionCondition: '47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU',
+  expiresAt: '2016-05-18T12:00:00.000Z',
+  data: Buffer.alloc(0),
+  custom: {}
+})
+```
+
+For a detailed description of these properties, please see [`Class: Transfer`](#class-transfer).
+
+#### LedgerPlugin#registerTransferHandler
+<code>ledgerPlugin.registerTransferHandler( **transferHandler**: ( transfer: [Transfer](#class-transfer) ) ⇒ Promise&lt;[Message](#class-message)> ) ⇒ null</code>
+
+Set the callback which is used to handle incoming prepared transfers. The callback should expect one parameter (the [Transfer](#class-transfer)) and return a promise for the resulting [FulfillmentInfo](#class-fulfillmentinfo). If the transfer is rejected or an error occurs, the callback should reject the transfer. In general, the callback should behave as [`sendTransfer`](#ledgerpluginsendtransfer) does.
+
+If a transfer handler is already set, this method throws a `TransferHandlerAlreadyRegisteredError`. In order to change the transfer handler, the old handler must first be removed via [`deregisterTransferHandler`](#ledgerpluginderegistertransferhandler). This is to ensure that handlers are not overwritten by accident.
+
+If an incoming transfer is received by the plugin, but no handler is registered, the plugin should reject the transfer.
+
+#### LedgerPlugin#deregisterTransferHandler
+<code>ledgerPlugin.deregisterTransferHandler( ) ⇒ null</code>
+
+Removes the currently used transfer handler. This has the same effect as if [`registerTransferHandler`](#ledgerpluginregistertransferhandler) had never been called.
+
+If no transfer handler is currently set, this method does nothing.
+
+### Event: `info_change`
+<code style="">ledgerPlugin.on('info_change',
+  (
+    **info**:[LedgerInfo](#class-ledgerinfo)
+  ) ⇒
+)</code>
+
+Emitted any time the plugin's `LedgerInfo` cache changes.
+
+## Class: Transfer
+<code>class Transfer</code>
+
+The `Transfer` class is used to describe transfers from the originator of the sendTransfer call towards some `destination`. All fields are required and MUST NOT be undefined. However, `amount` MAY be the value `'0'`, `data` MAY be an empty buffer and `custom` MAY be an empty object.
+
+###### Fields
+| Type | Name | Description |
+|:--|:--|:--|
+| `String` | [amount](#transferamount) | Integer transfer amount, in the ledger's base unit |
+| `String` | [destination](#transferdestination) | ILP destination address |
+| `Buffer` | [data](#transferdata) | Transport-layer data |
+| `String` | [executionCondition](#transferexecutioncondition) | Cryptographic hold condition |
```

Comment:
Minor nit, but I prefer `expiresAt`. 

Also, I like `executionCondition` because, in our implementation at least, we have other types of conditions floating around, and the naming is nice to be able to distinguish (I know this spec won't exactly match Atomic-mode implementations, but FWIW).

Reply:
> I like executionCondition because, in our implementation at least, we have other types of conditions floating around

Fair point

Reply:
Yeah for `executionCondition` I ran into the same issue - ilp-connector supports a cancellationCondition. For `expiresAt` - it's more consistent to say `expiresAt`, because we have other timestamps using that `...At` format, e.g. `triggeredAt`.

Don't think there is a strong reason to rename these fields. I was ready to do it, but @sharafian vetoed it and I'm kind of in the mode of, "let's just get this done," so if I can avoid some extra renaming, I'm happy.